plugins/tracing/http.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2017 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tracing
+
+import (
+	"net/http"
+	"net/http/httptrace"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	tlog "github.com/opentracing/opentracing-go/log"
+)
+
+// Transport wraps a RoundTripper. If a request is being traced with
+// Tracer, Transport will inject the current span into the headers,
+// and set HTTP related tags on the span.
+type Transport struct {
+	// The actual RoundTripper to use for the request. A nil
+	// RoundTripper defaults to http.DefaultTransport.
+	http.RoundTripper
+}
+
+// RoundTrip implements the RoundTripper interface.
+func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt := t.RoundTripper
+	if rt == nil {
+		rt = http.DefaultTransport
+	}
+
+	span, ctx := StartSpanFromContext(req.Context(), "HTTP: "+req.Method)
+	if span == nil {
+		return rt.RoundTrip(req)
+	}
+	defer span.Finish()
+
+	ht := tracer{span: span}
+	req = req.WithContext(httptrace.WithClientTrace(ctx, ht.clientTrace()))
```

Comment:
I'm hesitate to add this.  We should provide an option to enable/disable this for debugging, and http trace should be off by default.

Reply:
So why are you using `tracing http transport`?

Reply:
I mean enable/disable httptrace, not the whole tracing.  By default, we only trace http transaction, but user is able to turn on httptrace to debug (e.g. DNS issue) if needed.

It's debatable if we want to enable dns hook by default, but please do not turn on all httptrace hooks.